src/Storage/TSStorageManager.m

```diff
@@ -92,24 +110,52 @@ + (instancetype)sharedManager {
 - (instancetype)initDefault
 {
     self = [super init];
+    
+    if (![self tryToLoadDatabase]) {
+        // Failing to load the database is a catastrophic.
+        //
+        // The best we can try to do is to discard the current database
+        // and behave like a clean install.
+        
+        OWSAnalyticsCritical(@"Could not load database");
+    
+        // Try to reset app by deleting database.
+        [self deletePasswordFromKeychain];
```

Comment:
We are missing the call to `[TSAttachmentStream deleteAttachments];` which deletes any on-disk attachments.  Instead of these lines, `wipeSignalStorage` is *almost* what we want but won't work as is - though see below for my comment there.